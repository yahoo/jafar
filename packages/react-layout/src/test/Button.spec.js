/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import React from 'react';
import { shallow } from 'enzyme';
import Button from '../components/Button/index';

describe('Button', () => {
  let props;

  beforeEach(() => {
    props = {
      label: 'save',
      onClick: () => {},
    };
  });
  it('Should render - ok', () => {
    const component = shallow(getComponent(props));
    expect(component).toMatchSnapshot();
  });

  it('Should render - disabled truthy, render ok', () => {
    props.disabled = true;
    const component = shallow(getComponent(props));
    expect(component).toMatchSnapshot();
  });

  it('Should render with custom button component', () => {
    props.component = props => (<span>Custom</span>);
    const component = shallow(getComponent(props));
    expect(component).toMatchSnapshot();
  });

  function getComponent(props) {
    return (<Button {...props} />);
  }
});
