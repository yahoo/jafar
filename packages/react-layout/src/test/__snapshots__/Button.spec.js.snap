// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render - disabled truthy, render ok 1`] = `
<WithStyles(ForwardRef(Button))
  color="primary"
  disabled={true}
  onClick={[Function]}
>
  save
</WithStyles(ForwardRef(Button))>
`;

exports[`Button Should render - ok 1`] = `
<WithStyles(ForwardRef(Button))
  color="primary"
  onClick={[Function]}
>
  save
</WithStyles(ForwardRef(Button))>
`;

exports[`Button Should render with custom button component 1`] = `
<Component
  label="save"
  onClick={[Function]}
>
  save
</Component>
`;
